### DISCUSSION TITLE
Privacy and Data Sharing

### DISCUSSION THREAD

**Discussion by karkonium** [ocrmypdf/OCRmyPDF#1170] [Q&A] [answered]
Hi, I've been using OCRmyPDF and appreciate your focus on privacy. To confirm, does the OCRmyPDF application make any network requests or send data externally when/after processing PDF files? I've reviewed the source code and monitored network logs but wanted to double-check for added assurance.

**Comment by jbarlow83** [comment, marked as answer]
None whatsoever, unless you install a plugin that does. 

I have never tried, but you could even put it in a container (e.g. flatpak, lxc) and block network access, and everything will work without network if you want an extra level of assurance. 

Some details:
* The OCRmyPDF-EasyOCR plugin will automatically download language data if not installed.
* I can't speak for third party plugins.
* Possibly localhost sockets or UNIX sockets are not needed in some very specific situations and uncommon operating systems, because I don't know how how multiprocessing is implemented by Python on some obscure platforms, but there shouldn't be any need for outbound communication in any standard ocrmypdf configuration.

**Reply by karkonium** [reply]
Perfect, just what i was hoping for. Thank u!
